drake/solvers/trajectoryOptimization/PsmTraj.m

```diff
@@ -0,0 +1,445 @@
+classdef PsmTraj < NonlinearProgram
+  % Pseudospectral method
+  % Knot pointes are determined by N-th order orthogonal polynomials, e.g. Legendre polynomials, Chebyshev polynomials
+  % Then at each knot points, use Lagrange polynomials to approximate state values. 
+  % f(x(k),u(k)) and f(x(k+1),u(k+1)) are evaluated at the knot points to match the derivative of the Lagrange poly
+   %%options,'ps_method' detemines if to use Legendre (option.ps_method=1) or to use Chebyshev (option.ps_method=2)
+
+  properties (SetAccess = protected)
+    N       % number of timesteps
+    options % options, yup
+    plant   % the plant
+    h_inds  % (N-1) x 1 indices for timesteps h so that z(h_inds(i)) = h(i)
+    x_inds  % n x N indices for state
+    u_inds  % m x N indices for input
+    tau
+    D
+    w
+    dynamic_constraints = {};
+    constraints = {};
+  end
+  
+  methods
+    function obj = PsmTraj(plant,N,durations,options)
+      if isscalar(durations), durations=[durations,durations]; end
+      if nargin < 4
+        options = struct();
+      end
+      if ~isfield(options,'ps_method')
+        options.ps_method = 1;
```

Comment:
Use string 'chebyshev' or 'legendre' instead of value '1' or '2'. Or you may derive two classes `ChebyshevPseudoSpectralTraj` and `LegendrePseudoSpctralTraj` from `PsmTraj`